### PR TITLE
Pin version of ElasticSearch client to 7.13.0

### PR DESCRIPTION
**Pull request by Xantier**
## Hey, I just made a Pull Request!

Pinning ElasticSearch client to 7.13.0 so that third party clusters can be used. Elastic introduced functionality to break AWS OpenSearch on 7.14.0. With this change we can continue supporting AWS OpenSearch without a need to externally enforce client versions. Likely need to be revisited in the future if there is some functionality needed that we'd receive by updating the client.

Refs:
* https://aws.amazon.com/blogs/opensource/keeping-clients-of-opensearch-and-elasticsearch-compatible-with-open-source/
* https://github.com/backstage/backstage/issues/7839

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [n/a] Added or updated documentation
- [n/a] Tests for new functionality and regression tests for bug fixes
- [n/a] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

**Review by emmaindal — APPROVED** [review]
Looks good to me, I wonder if we should open a issue to track 

"Likely need to be revisited in the future if there is some functionality needed that we'd receive by updating the client."

or do you think https://github.com/backstage/backstage/issues/7839 is enough (and should be left open) if we just update it a bit?

**Comment by freben** [comment]
Good call - yeah maybe rewording the existing one slightly would make sense.
